src/diart/models.py

```diff
@@ -45,14 +59,38 @@ def __call__(self, *args, **kwargs):
         return super().__call__(*args, **kwargs)
 
 
+class LazyWeSpeakerSpeakerEmbedding(WeSpeakerPretrainedSpeakerEmbedding, ABC):
+    def __init__(self, loader: Callable[[], WeSpeakerPretrainedSpeakerEmbedding]):
+        self.get_inference = loader
+        self.inference: Optional[WeSpeakerPretrainedSpeakerEmbedding] = None
+        # __init__ at end because in there we call .to() which requires .load()
+        super().__init__()
+
+    def is_in_memory(self) -> bool:
+        """Return whether the model has been loaded into memory"""
+        return self.inference is not None
+
+    def load(self):
+        if not self.is_in_memory():
+            self.inference = self.get_inference()
+
+    def to(self, *args, **kwargs) -> WeSpeakerPretrainedSpeakerEmbedding:
+        self.load()
+        return super().to(*args, **kwargs)
+
+    def __call__(self, *args, **kwargs):
+        self.load()
+        return super().__call__(*args, **kwargs)
+
+
 class SegmentationModel(LazyModel):
     """
     Minimal interface for a segmentation model.
     """
 
     @staticmethod
     def from_pyannote(
-        model, use_hf_token: Union[Text, bool, None] = True
+            model, use_hf_token: Union[Text, bool, None] = True
```

Comment:
plase make sure to run `black src/diart` so all code is well-formatted.

Reply:
I ran `black src/diart`, this also changed `src/diart/console/stream.py` which I did not modify. Should I revert that change or keep it?

Reply:
that's ok the other changes can be kept, I must have forgotten to run it myself